### ISSUE TITLE
Adding just Microsoft.ProjectReunion does not bring in all the required MSBuild targets

### ISSUE THREAD

**Issue by mattleibow** [microsoft/WindowsAppSDK#977]
**Describe the bug**

Many of the Microsoft.ProjectReunion.* NuGet packages do not include the targets files again in the `buildTransitive`. This means that if I just add the `Microsoft.ProjectReunion` package to a WinUI app, the other targets from .Foundation and .WinUI are not included. As a result, the app does not package correctly.

**Steps to reproduce the bug**

1. File | New WinUI project using the new Single Project MSIX system
2. Remove the WinUI packages, leaving the "root" `Microsoft.ProjectReunion`
3. Observe that you can't run the app

**Expected behavior**
The app builds the same whether I include all the Microsoft.ProjectReunion packages or not.

**Screenshots**

**Version Info**

NuGet package version: 

 - Microsoft.ProjectReunion 0.8.0-preview

**Additional context**

**Comment by BenJKuhn** [comment]
This is by design. The separate Nuget packages are an artifact of how we built up the project. They are not intended nor designed to be used in isolation.

**Comment by dotMorten** [comment]
@BenJKuhn I don't think this is correct.
If I have class library, which references the Project Reunion, Foundation and WinUI packages, and I then have an application that reference Class Library A, it would fail to run, because it doesn't implicitly include all the required Windows App SDK stuff that is needed. The user has to magically know to also reference the windows app sdk packages explicitly.

This is also in part why I earlier on raised an issue about reunion referencing WinUI, instead of the other way around, because it messes up this flow up quite a lot (ie my class library might depend on winui, so that's what I reference, and in turn winui depends on reunion, so in turn it references that, but today it is the other way around).

**Comment by mattleibow** [comment]
I agree with @dotMorten here. Right now we are basically adding this to all our templates because WinUI will not build a valid app unless the packages are installed directly. 

```xml
	<ItemGroup Condition="$(TargetFramework.Contains('-windows'))">
		<PackageReference Include="Microsoft.ProjectReunion" Version="0.8.1" />
		<PackageReference Include="Microsoft.ProjectReunion.Foundation" Version="0.8.1" />
		<PackageReference Include="Microsoft.ProjectReunion.WinUI" Version="0.8.1" />
		<PackageReference Include="Microsoft.ProjectReunion.InteractiveExperiences" Version="0.8.1" NoWarn="NU1701" />
	</ItemGroup>
```

**Comment by mattleibow** [comment]
@BenJKuhn I am not saying that the packages should all be included, I am saying that when I reference a package, I want it to actually be included.

Right now we have this:

 - .NET 6 App
   - App Framework
	 - WinUI

Since the App Framework depends on WinUI, I would expect that the .NET 6 App also gets the targets. It gets the dlls so I can write some code, but then blows up because the targets were not also included.

**Comment by dotMorten** [comment]
> This is by design. The separate Nuget packages are an artifact of how we built up the project.

I'm finding it rather trouble some that "how you build the project" affects all customers who tries to use the product. IMHO that just means "how you build it" isn't right. I can actually go and manually fix these packages by hand, so why can't a post-build script?

**Comment by dotMorten** [comment]
> This is by design. The separate Nuget packages are an artifact of how we built up the project. They are not intended nor designed to be used in isolation.

@BenJKuhn @andrewleader Now that it is all a single package, can we address you lacking support for transitive builds now? 

We're seeing [stuff like this](https://github.com/dotnet/maui/blob/main/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj#L48-L54) in projects. We can't just reference a nuget package (like Maui) that takes a dependency on the WinAppSDK, but have to magically know to also add an explicit reference to the WinAppSDK in every single project in the project graph. This is why NuGet has transitive build support. I don't really get from the above thread at all why the WinAppSDK can't make use of this?

**Comment by andrewleader** [comment]
Fully agreed @dotMorten, this should just work. I've created #1469 to track this. I'll try to get a cost estimate and maybe we can get this fixed for 1.0, but no promises for 1.0.
